### PR TITLE
[Set] Add new rector-preset set to have opiniated rector rules that can be enabled in rector-src and its extension

### PR DESCRIPTION
**Pull request by samsonasik**
Alternative of https://github.com/rectorphp/rector-src/pull/5923
Closes https://github.com/rectorphp/rector-src/pull/5923

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba you choose ;)

I prefer to this PR, so no need repetitive copy paste rules in rector-src and its extension for our standard so they are actually tested in real our code base.

File: rules/TypeDeclaration/Rector/StmtsAwareInterface/IncreaseDeclareStrictTypesRector.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function beforeTraverse(array $nodes): ?array
         $rectorWithLineChange = new RectorWithLineChange(self::class, $stmt->getLine());
         $this->file->addRectorClassWithLine($rectorWithLineChange);
 
-        return \array_merge([$strictTypesDeclare, new Nop()], $nodes);
+        return [$strictTypesDeclare, new Nop(), ...$nodes];
```
Comment (TomasVotruba) [inline rules/TypeDeclaration/Rector/StmtsAwareInterface/IncreaseDeclareStrictTypesRector.php]: This rule should be excluded, as it makes code harder to read :+1:

**Comment by TomasVotruba** [comment]
Looks good, just one rule to ommit and its ready :+1:

File: rules/TypeDeclaration/Rector/StmtsAwareInterface/IncreaseDeclareStrictTypesRector.php (hunk 1 above)
Comment (samsonasik) [reply rules/TypeDeclaration/Rector/StmtsAwareInterface/IncreaseDeclareStrictTypesRector.php]: sure, excluded https://github.com/rectorphp/rector-src/pull/5924/commits/fa0e1d48d4dc53359806080effb1063f998ea95c

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I am merging it ;)

**Comment by TomasVotruba** [comment]
Awesome :)

**Comment by andrew-demb** [comment]
What about adding a short description for the `SetList` constants?
@samsonasik 

I searched the reason why I have this warning in the output:
```
 [WARNING] Skipped rule                                                         

           "Rector\CodingStyle\Rector\PostInc\PostIncDecToPreIncDecRector" is  never registered. You can remove it from "->withSkip()"              
```

In the codebase, I found that this rule is now enabled only in `rector-preset`, but I didn't know whether it is a new set or what the semantics of it

**Comment by samsonasik** [comment]
@andrew-demb good idea, could you add PR for it? It basically opiniated rules that match rector coding standard

**Pull request merged**